File: service-one-individual-discount/ports/mongodb_port.js
var dbName = "TestDB";
var MongoClient = require('mongodb').MongoClient;
var url = `mongodb://localhost:27017/${dbName}`;

/**
 * Creates a collection on the database with the name received
 * @param {String} collectionName 
 * @param {String} dbName 
 */
function createCollection(collectionName, dbName) {
    return new Promise((resolve, reject) => {
        MongoClient.connect(url, function(err, db) {
            if (err) throw reject(err);
            var dbo = db.db(dbName);
            dbo.createCollection(collectionName, function(err, res) {
              if (err) throw reject(err);
              db.close();
              resolve("Collection created!");      
            });
        });
    });
}

/**
 * Inserts a object into the collection and data base received
 * @param {Object} data 
 * @param {String} collectionName 
 * @param {String} dbName 
 */
function insert(data, collectionName, dbName) {
    return new Promise((resolve, reject) => {
        MongoClient.connect(url, function(err, db) {
            if (err) throw reject(err);
            var dbo = db.db(dbName);
            dbo.collection(collectionName).insertOne(data, function(err, res) {
              if (err) throw reject(err);
              db.close();
              resolve("1 document inserted");
            });
        });
    });
    
}

/**
 * Find a object from the collection and data base received using the query object
 * @param {Object} query 
 * @param {String} collectionName 
 * @param {String} dbName
 * @returns {Object} result 
 */
function findOne(query, collectionName, dbName) {
    return new Promise((resolve,reject) => {
        MongoClient.connect(url, function(err, db) {
            if (err) throw reject(err);
            var dbo = db.db(dbName);
            dbo.collection(collectionName).findOne(query, function(err, result) {
              if (err) throw reject(err);
              db.close();
              resolve(result);
            });
        });
    });
}

/**
 * Return all objects from the data base and collection received
 * @param {String} collectionName 
 * @param {String} dbName
 * @returns {Array} result
 */
function findAll(collectionName, dbName) {
    return new Promise((resolve, reject) => {
        MongoClient.connect(url, function(err, db) {
            if (err) throw reject(err);
            var dbo = db.db(dbName);
            dbo.collection(collectionName).find({}).toArray(function(err, result) {
              if (err) throw reject(err);
              db.close();
              resolve(result);
            });
        });
    });
}

module.exports = {
    createCollection: createCollection,
    insert: insert,
    findOne: findOne,
    findAll: findAll
}

File: service-two-product-listing/mongodb_port.py
import pymongo
import json

myclient = pymongo.MongoClient("mongodb://localhost/TestCollection")
mydb = myclient["TestDB"]
mycol = mydb["testUsersCollection"]

def insert(data, collectionName, dbName):
    """Inserts the data dict into the collection and data base received

    Args:
        arg1 (dict): Data to be inserted
        arg2 (string): Collection Name
        arg3 (string): Data Base Name

    """
    myclient[dbName][collectionName].insert_one(data)

def findOne(query, collectionName, dbName):
    """Returns a dict from the collection, database and query provided.

    Args:
        arg1 (dict): Query data
        arg2 (string): Collection Name
        arg3 (string): Data Base Name
    Returns:
        dict: Dict from Data Base
    """
    return myclient[dbName][collectionName].find(query)

def findAll(collectionName, dbName):
    """Returns a list of all objects from the collection and database provided.

    Args:
        arg1 (dict): Query data
        arg2 (string): Collection Name
        arg3 (string): Data Base Name
    Returns:
        list: List of objects from Data Base
    """
    products = myclient[dbName][collectionName].find({})
    response = []
    for product in products:
        del product["_id"]
        response.append(product)
    return response